pkg/flowaggregator/flowaggregator.go

```diff
@@ -15,25 +15,263 @@
+	obsID, err := genObservationID()
+	if err != nil {
+		return fmt.Errorf("cannot generate obsID for IPFIX ipfixexport: %v", err)
+	}
+	var ep ipfix.IPFIXExportingProcess
+	if fa.externalFlowCollectorAddr.Network() == "tcp" {
+		// TCP transport do not need any tempRefTimeout, so sending 0.
+		ep, err = ipfix.NewIPFIXExportingProcess(fa.externalFlowCollectorAddr, obsID, 0)
+	} else {
+		// For UDP transport, hardcoding tempRefTimeout value as 1800s.
+		ep, err = ipfix.NewIPFIXExportingProcess(fa.externalFlowCollectorAddr, obsID, 1800)
+	}
+	if err != nil {
+		return fmt.Errorf("Got error when initializing IPFIX exporting process: %v", err)
+	}
+	fa.exportingProcess = ep
+	fa.templateID = fa.exportingProcess.NewTemplateID()
+	templateSet := ipfix.NewSet(ipfixentities.Template, fa.templateID, false)
+
+	bytesSent, err := fa.sendTemplateRecord(templateSet)
+	if err != nil {
+		fa.exportingProcess = nil
+		fa.templateID = 0
+		return fmt.Errorf("Send template message failed, err: %v", err)
 	}
+	klog.V(2).Infof("Initialized exporting process and sent %d bytes size of template record", bytesSent)
+	return nil
 }
 
 func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
-	// Place holder for Run() function, logic will be added in next PR
+	var collectAddr net.Addr
+	var cp ipfix.IPFIXCollectingProcess
+	var err error
+	if fa.collectingProtocol == "tcp" {
+		collectAddr, _ = net.ResolveTCPAddr("tcp", "0.0.0.0:4739")
+		cp, err = ipfix.NewIPFIXCollectingProcess(collectAddr, 65535, 0)
+	} else {
+		collectAddr, _ = net.ResolveUDPAddr("udp", "0.0.0.0:4739")
+		cp, err = ipfix.NewIPFIXCollectingProcess(collectAddr, 1024, 0)
+	}
+	if err != nil {
+		klog.Errorf("Error when create collecting process: %v", err)
+		return
+	}
+	fa.collectingProcess = cp
+
+	ap, err := ipfix.NewIPFIXAggregationProcess(fa.collectingProcess.GetMsgChan(), 2, aggregationFields)
+	if err != nil {
+		klog.Errorf("Error when create aggreagation process: %v", err)
+		return
+	}
+	fa.aggregationProcess = ap
+
+	exportTicker := time.NewTicker(fa.exportInterval)
+	defer exportTicker.Stop()
+	go fa.collectingProcess.Start()
+	go fa.aggregationProcess.Start()
+	for {
+		select {
+		case <-stopCh:
+			fa.collectingProcess.Stop()
+			fa.aggregationProcess.Stop()
+			if fa.exportingProcess != nil {
+				fa.exportingProcess.CloseConnToCollector()
+			}
+			return
+		case <-exportTicker.C:
+			if fa.exportingProcess == nil {
+				err := fa.initExportingProcess()
+				if err != nil {
+					klog.Errorf("Error when initializing exporting process: %v", err)
+					// There could be other errors while initializing exporting process other than connecting to IPFIX collector,
+					// therefore closing the connection and resetting the process.
+					if fa.exportingProcess != nil {
+						fa.exportingProcess.CloseConnToCollector()
+						fa.exportingProcess = nil
+					}
+					continue
+				}
+			}
+			err := fa.aggregationProcess.ForAllRecordsDo(fa.sendFlowKeyRecords)
```

Comment:
if the number of records is quite high, you may want to pre-allocate memory for the `fa.sentFlowKeys` slice